bigquery/storage/managedwriter/managed_stream.go

```diff
@@ -327,15 +331,29 @@ func (ms *ManagedStream) Close() error {
 // AppendRows sends the append requests to the service, and returns a single AppendResult for tracking
 // the set of data.
 //
-// The format of the row data is binary serialized protocol buffer bytes, and and the message
-// must adhere to the format of the schema Descriptor passed in when creating the managed stream.
-func (ms *ManagedStream) AppendRows(ctx context.Context, data [][]byte, offset int64) (*AppendResult, error) {
+// The format of the row data is binary serialized protocol buffer bytes.  The message must be compatible
+// with the schema currently set for the stream.
+//
+// Use the sentinel value NoStreamOffset to omit sending of the offset value.
```

Comment:
From my testing, they must send an offset if using PENDING mode. And they can't if using default stream, right? I didn't test with BUFFERED, so maybe it's optional there? Maybe there's a better way to communicate when NoStreamOffset should be set?

Reply:
Interestingly enough, this gets to the other breaking change I've been considering here.  Remove offset as a static argument from the AppendRows() function and add a `WithOffset()` AppendOption.

The origin of the `NoStreamOffset` was to simplify the AppendResult, as Go's lack of null vs default value makes it more complex to deal with the optional offset.  I didn't want to do `*int64`, but it's an option to consider as well.

Reply:
Given this is already a kind of managed writer i wonder if this cannot be abstracted away completely? But I guess to not force how to handle errors in a retry-able manner it is needed to expose it? Either way, I do not find the NoStreamOffset sentient value a big deal, works fine.

Reply:
Went ahead and made the offset part of the variadic options.

This changes the signatures for appending to:
* No offset set: `<ManagedStream>.AppendRows(ctx, data)`
* Offset set: `<ManagedStream>.AppendRows(ctx, data, WithOffset(offset))`